src/ucp/core/ucp_am.c

```diff
@@ -328,6 +328,10 @@ ucs_status_t ucp_worker_set_am_recv_handler(ucp_worker_h worker,
         return UCS_ERR_INVALID_PARAM;
     }
 
+    if (param->id & 0xffff0000) {
```

Comment:
minor: maybe `if (param->id != (uint16_t)param->id)` to make it more explicit?

Reply:
`if (param->id > UINT16_MAX)`